### ISSUE TITLE
[component name |ui5WebComponents | 1.13.0  | icons | theming | localization | i18n | typescript]:

### ISSUE THREAD

**Issue by ShivamGuptaSap** [SAP/ui5-webcomponents#7025]
### Bug Description

After upgrading to the latest ui5-WebComponents version 1.13.0, and ui5-webComponents-react to 1.15.1 in SAP LWV Project.

This Ui5-WebComponents-base version 1.13.0 is throwing the below mentioned error and we need this  ui5-WebComponents 
 1.13.0 version as it contains fix for previously raised bugs. 

<img width="1104" alt="Screenshot 2023-05-11 at 3 01 33 PM" src="https://github.com/SAP/ui5-webcomponents/assets/107616257/acf61a39-0a13-4c96-8aec-77a6b7c06496">

Also, we have tried the lower versions of ui5-webcomponents-base like 1.12.2
with that version we are not getting the above error but in that case the default theme is getting impacted.

Note: After the update we are not able to see any thing on the screen.

### Affected Component

Toolbar

### Expected Behaviour

After upgrading to the latest versions, app should run like before and there should not be any errors in the console or terminal.

### Steps to Reproduce

1.
2.
3.
...

### Priority

None

### UI5 Web Components Version

1.13.0

### Browser

Chrome

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by ShivamGuptaSap** [comment]
Hello @niyap,
This issue is not reproducible in the sample codeSandbox but if you check the screenshot which is attached above, there after we upgrade @ui5/webcomponents-base version 1.13.0 in Liteviewer Project.
It throws error like cannot read property of null (reading 'insertBefore')  in Runtime.js, Render.js etc which are internal files present in  @ui5/webcomponents-base/dist.

**Comment by ilhan007** [comment]
Hello @ShivamGuptaSap thank you for the input, indeed we can see that error is thrown by our code - from getSingletonElementInstance method, we just wanted to get as much info as possible to be able to reproduce the issue, once we reproduce the issue, we can work on a fix.

As, we don't have access to your application, and we don't have a sample to see the issue, we looked at changes that could be related and relatively recent and this one came up: https://github.com/SAP/ui5-webcomponents/pull/6848

So could you try to revert the change and get back to us with the result?

To revert it on your side you can change the file in the node_modules and run your app locally:
1. Step 1- find the `node_modules/@ui5/webcomponents-base/dist/getSharedResource.ts` file

2. Step 2- perform a change in one line

change
```ts
return getSingletonElementInstance("ui5-shared-resources")
```
to (providing a 2nd parameter)
```ts
return getSingletonElementInstance("ui5-shared-resources", document.head)
```



BR,
ilhan

**Comment by ShivamGuptaSap** [comment]
Hi @ilhan007, 
I checked the above mentioned change in getSharedResource.ts file with UI5Wbcomponents version 1.13.2 .
It worked, Kindly help us by providing this change as soon as possible, So we can consume it in our application.
Thanks 
Shivam Gupta

**Comment by ilhan007** [comment]
Hello @ShivamGuptaSap ok, so we definitely know the change is an issue for you

Could you try one more thing?

Step 1 -  find the same file (as it is without the fix)  `node_modules/@ui5/webcomponents-base/dist/getSharedResource.ts`

Step 2 - add `console.log("Document state : " + document.readyState)` just before the return statement:

```js
console.log("Document state : " + document.readyState)
return getSingletonElementInstance("ui5-shared-resources") as unknown as Record<string, unknown>;
```

and tell us if it's printing "`loading`" in the console, just before the error. If it's "`loading`" then it would mean the body is not ready yet and can be accessed via `document.body`.


The easiest thing is to move the element back to the **head**, where the "ui5-shared-resources" html element used to be created previously. However, we had issue that only restricted list of elements can be appended to the head, and that's why we moved it to the **body**. But, it seems the document.body is not ready in your case, so probably the fix would be to wait until it's ready.

BR,
ilhan

**Comment by ilhan007** [comment]
Hello @ShivamGuptaSap 

Also, it would be also useful if you provide a link to your app or app's repository, so we can clone your app and run it locally, so that we can test solutions using it.

It will be quicker this way.

The GitHub is public, but you can find me via in internal SAP channels, because I assume you are a SAP Developer, if I am not mistaking.

ilhan

**Comment by Abanindra** [comment]
Hi @ilhan007 What's the best way to collaborate here with you - through Slack or MS Team?

**Comment by ilhan007** [comment]
@Abanindra both are fine

**Comment by Abanindra** [comment]
Hi @ilhan007 Thanks for your quick response here. We will be trying out your suggestion and let you know if this works or not.

**Comment by Abanindra** [comment]
Hi @ilhan007 I hope you have received an update from @ShivamGuptaSap . Please suggest what's the next step here as we are kind of block to upgrade to UI5WC V 1.13.2?

**Comment by ilhan007** [comment]
Hello @Abanindra no I haven't received anything yet.

**Comment by Abanindra** [comment]
Hi @ShivamGuptaSap Can you please share the update?

**Comment by ShivamGuptaSap** [comment]
Hi @ilhan007,
I've tried out console.log("Document state : " + document.readyState) just before the return statement and it's printing "loading" in the console, just before the error.
Also I pinged you over slack yesterday regarding this. 
![image](https://github.com/SAP/ui5-webcomponents/assets/107616257/750499df-1d40-44d0-83ab-83fa2352e05e)

**Comment by Abanindra** [comment]
Hi @ilhan007 Looks like Shivam has shared the same update through Slack to you yesterday. We need an early resolution of this issue, so please suggest if we can have a quick call sometime today?

**Comment by ilhan007** [comment]
Resolved with: 1.13.3
